_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,392 @@
+
+The following configuration options are **required** in order to build a connector:
+
+| Field | Data type | Description |
+| :---  | :--- | :--- |
+| `name` | String | The name of the connector. |
+| `description` | String | The description for the connector. |
+| `version` | Integer | The version for the connector. |
+| `protocol` | String | The protocol for the connection. For AWS services such as SageMaker and Bedrock, use `aws_sigv4`. For all other services, use `http`. |
+| `parameter` | JSON array | The default connector parameters, including `endpoint` and `model`. 
+| `credential` | String | Defines any credential variables required connect to your chosen endpoint. ML Commons uses **AES/GCM/NoPadding** symmetric encryption with a key length of 32 bytes. When a connection cluster first starts, the key persists inside of OpenSearch. Therefore, you do not need to manually encrypt the key.
+| `action` | JSON array | Tells the connector what actions to run after a connection to ML Commons has been made. For more information about how to configure actions, see [Actions](#action-settings).
+
+### Action settings
+
+The `action` setting when creating a connection tells the connector what ML Commons API operation to run against the connection endpoint. You can configure actions using the following settings:
+
+| Field | Data type | Description |
+| :---  | :--- | :--- |
+`action_type` | String | Required. Sets the ML Commons API operation to use upon connection. As of OpenSearch 2.9, only `predict` is supported. 
+`method` | String | Required. Defines the HTTP method for the API call. Supports `POST` and `GET`.
+`url` | String | Required. Sets the connection endpoint the action takes place. This must match the regex expression for the connection using when [adding trusted endpoints](#adding-trusted-endpoints).
+`headers` | Sets the headers used inside the request or response body. Default is `application/json`.
+`request_body` | Required. Sets the parameters contained inside the request body of the action.
+
+## Registering and deploying a connected model
+
+After a connection has been created, use the `connector_id` from the response to register and deploy a connected models.
+
+For registering model, you have the following options:
+
+- You can use `model_group_id` to create the model with an associated model group.
+- If you do not use `model_group_id`, a new model is created.
+
+The following example registers a model named `openAI-GPT-3.5 completions`:
+
+```
+POST /_plugins/_ml/models/_register
+{
+    "name": "openAI-GPT-3.5 completions",
+    "function_name": "remote",
+    "model_group_id": "lEFGL4kB4ubqQRzegPo2",
+    "description": "test model",
+    "connector_id": "kUFFL4kB4ubqQRzeQ_r-"
+}
+```
+
+ML Commons returns the `task_id` and registration status of the model. You can use the `task_id` to find the `model_id`, as showing the following example:
+
+**GET model request**
+
+```
+GET /_plugins/_ml/tasks/l0FIL4kB4ubqQRzeKvpV
+```
```

Comment:
Please make sure to use json for all requests and add copy buttons where needed.

Reply:
Do have a distinction with the copy button between illustrative examples that are meant to guide the user but not be used directly, like this one or most example responses, and examples that are meant to be copied and replaced with a user's custom values?

Reply:
Copy buttons copy all text. Users can replace what needs to be replaced directly but there is no functionality for such replacement through the UI